internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -237,82 +444,496 @@ func (s *Server) HandleGitHubWebHook() http.HandlerFunc {
 			Str("providertype", m.Metadata[events.ProviderTypeKey]).
 			Str("upstream-delivery-id", m.Metadata[events.ProviderDeliveryIdKey]).
 			Logger()
+		ctx = l.WithContext(ctx)
 
 		l.Debug().Msg("parsing event")
 
-		// Parse the webhook event and construct a message for the event router
-		if err := s.parseGithubEventForProcessing(rawWBPayload, m); err != nil {
-			wes = handleParseError(wes.Typ, err)
+		wes.Accepted = true
+		var res *processingResult
+		var processingErr error
+
+		// The following two switch statements are effectively
+		// mutually exclusive and the set of events that they
+		// manage is non-overlapping.
+		//
+		// This is not verified statically and it is not yet
+		// verified via tests, but should be easy enough to
+		// verify by inspection.
+
+		switch github.WebHookType(r) {
+		// All these events are related to a repo and usually
+		// contain an action. They all trigger a
+		// reconciliation or, in some cases, a deletion.
+		case "repository", "meta":
+			res, processingErr = s.processRelevantRepositoryEvent(ctx, rawWBPayload)
+		case "branch_protection_configuration",
+			"branch_protection_rule",
+			"code_scanning_alert",
+			"create",
+			"member",
+			"public",
+			"push",
+			"repository_advisory",
+			"repository_import",
+			"repository_ruleset",
+			"repository_vulnerability_alert",
+			"secret_scanning_alert",
+			"secret_scanning_alert_location",
+			"security_advisory",
+			"security_and_analysis",
+			"team",
+			"team_add":
+			res, processingErr = s.processRepositoryEvent(ctx, rawWBPayload)
+		case "package":
+			// This is an artifact-related event, and can
+			// only trigger a reconciliation.
+			res, processingErr = s.processPackageEvent(ctx, rawWBPayload)
+		case "pull_request":
+			res, processingErr = s.processPullRequestEvent(ctx, rawWBPayload)
+		// This event is not currently handled.
+		case "org_block",
```

Comment:
I know we had them in the earlier list, but I can't tell why now. Since we don't handle them should we set accepted=false?

Reply:
I agree on setting accepted=false, I'll fix that.

That said, I had a look back at github documentation and it looks like these events do contain a repository, but their semantics is somewhat unrelated to it
* `org_block` event occurs when organization owners or moderators block or unblock a non-member from collaborating on the organization's repositories
* `organization` event occurs when there is activity relating to an organization and its members

I think we should not process them, but I am ok adding them back.

Reply:
I'm fine with removing. I don't remember why we added these events. I think in the distant past we handled all events and when we added the filtering we didn't want to filter too much. But I don't see a use for these events either.